src/main/scala/temple/generate/target/openapi/OpenAPIType.scala

```diff
@@ -0,0 +1,33 @@
+package temple.generate.target.openapi
+
+import io.circe.Json
+import io.circe.syntax._
+import temple.generate.target.openapi.OpenAPIType.{OpenAPIArray, OpenAPIObject}
+import temple.utils.MonadUtils.MatchPartial
+
+import scala.collection.immutable.ListMap
+
+sealed abstract private[openapi] class OpenAPIType(fieldEntries: Seq[(String, Json)]) extends Jsonable {
+  def typeString: String
+  final lazy val customFields: Map[String, Json] = fieldEntries.to(ListMap)
+
+  override def toJsonMap: Map[String, Option[Json]] =
+    ListMap(
+      "type"       -> Some(typeString.asJson),
+      "properties" -> this.matchPartial { case openAPIType: OpenAPIObject => openAPIType.properties.asJson },
+      "items"      -> this.matchPartial { case openAPIType: OpenAPIArray => openAPIType.items.asJson },
+    ) ++ customFields.view.mapValues(Some(_))
+}
+
+private[openapi] object OpenAPIType {
+
+  sealed case class OpenAPISimpleType(typeString: String, fieldEntries: (String, Json)*)
+      extends OpenAPIType(fieldEntries)
```

Comment:
What is `typeString` in this case where it isn't defined? How does the `toJsonMap` method work in this case?

Reply:
Anything like number or string, e.g. in

https://github.com/TempleEight/temple/blob/5c637f8d7ed6ae7d72d0acfe30891632e924958f/src/main/scala/temple/generate/target/openapi/OpenAPIGenerator.scala#L16

Reply:
In terms of how it works, it’s been updated slightly, but originally the definition of a value in a case class overrides a def in the parent class